### PR TITLE
refactor(workflows): Update code scanning workflow

### PR DESCRIPTION
**Pull request by ScottBrenner**
### Summary

Updating the code scanning workflow here, copied from https://github.com/actions/starter-workflows/blob/main/code-scanning/codeql.yml. Hoping to address errors like https://github.com/ScottBrenner/cfn-lint-action/runs/3721798387#step:3:53

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -1,38 +1,71 @@
...
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:11]:
```suggestion
# For most projects, this workflow file will not need changing; you simply need
# to commit it to your repository.
#
# You may wish to alter this file to override the set of languages analyzed,
# or to provide custom queries or build logic.
#
# ******** NOTE ********
# We have attempted to detect the languages in your repository. Please check
# the `language` matrix defined below to confirm you have the correct set of
# supported CodeQL languages.
#
```

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -2,37 +2,59 @@ name: "CodeQL"
...
 on:
   push:
-    branches-ignore: dependabot/**
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ $default-branch ]
   schedule:
-    - cron: '38 18 * * 3'
+    - cron: $cron-weekly
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:10]:
```suggestion
    - cron: '38 18 * * 3'
```

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -2,37 +2,59 @@ name: "CodeQL"
 
 on:
   push:
-    branches-ignore: dependabot/**
+    branches: [ $default-branch, $protected-branches ]
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:5]:
```suggestion
```

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -2,37 +2,59 @@ name: "CodeQL"
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ $default-branch ]
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:8]:
```suggestion
```
Not sure if we need to specify branches here, or maybe it should be `main`?

File: .github/workflows/codeql-analysis.yml [hunk 5]
```diff
@@ -2,37 +2,59 @@ name: "CodeQL"
...
+      actions: read
       contents: read
-      pull-requests: read
       security-events: write
 
     strategy:
       fail-fast: false
       matrix:
-        language: [ 'javascript' ]
+        language: [ $detected-codeql-languages ]
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:22]:
```suggestion
        language: [ 'javascript' ]
```
?

File: .github/workflows/codeql-analysis.yml [hunk 6]
```diff
@@ -2,37 +2,57 @@ name: "CodeQL"
 
 on:
   push:
-    branches-ignore: dependabot/**
+  pull_request:
+    # The branches below must be a subset of the branches above
```
Comment (ScottBrenner) [inline .github/workflows/codeql-analysis.yml:6]:
```suggestion
```

**Comment by NickLiffen** [comment]
@ScottBrenner why are we adding all the comments back in 🤔

File: .github/workflows/codeql-analysis.yml [hunk 7]
```diff
@@ -2,37 +2,56 @@ name: "CodeQL"
...
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
```
Comment (NickLiffen) [inline .github/workflows/codeql-analysis.yml:55]: This shouldn't be needed 🤔

**Review by NickLiffen — APPROVED** [review]

**Pull request merged**